data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3ObjectWorker.java

```diff
@@ -56,6 +57,10 @@ public S3ObjectWorker(final S3ObjectRequest s3ObjectRequest) {
     }
 
     public void parseS3Object(final S3ObjectReference s3ObjectReference) throws IOException {
```

Comment:
Same as other comment below, this method is never called. You can remove this and refactor tests.